theano/tensor/nnet/nnet.py

```diff
@@ -2014,7 +2014,8 @@ def binary_crossentropy(output, target):
     TODO : Rewrite as a scalar, and then broadcast to tensor.
 
     """
-    return -(target * tensor.log(output) + (1.0 - target) * tensor.log(1.0 - output))
+    one = np.asarray(1.0).astype(theano.config.floatX)
```

Comment:
I don't think this is a numpy issue. I won't investigate more if it is fixed on your side.